src/charm.py

```diff
@@ -72,13 +78,42 @@ def main(self, event):
 
         self.model.unit.status = ActiveStatus()
 
+    def _check_dex_oidc_config_relation(self) -> None:
+        """Check for exceptions from the library and raises ErrorWithStatus to set the unit status.
+
+        Raises:
+            ErrorWithStatus: if the relation hasn't been established, set unit to BlockedStatus
+            ErrorWithStatus: if the relation has empty or missing data, set unit to WaitingStatus
+        """
+        try:
+            self._dex_oidc_config_requirer.get_data()
+        except DexOidcConfigRelationMissingError as rel_error:
+            raise ErrorWithStatus(
+                f"{rel_error.message} Please add the missing relation.", BlockedStatus
+            )
+        except DexOidcConfigRelationDataMissingError as data_error:
+            self.logger.error(f"Empty or missing data. Got: {data_error.message}")
+            raise ErrorWithStatus(
+                f"Empty or missing data in {OIDC_PROVIDER_INFO_RELATION} relation."
+                " This may be transient, but if it persists it is likely an error.",
+                WaitingStatus,
+            )
+
+    @property
+    def _dex_oidc_config_requirer(self) -> DexOidcConfigRequirer:
+        """Return a DexOidcConfigRequirer object that gathers data from an OIDC provider."""
+        return DexOidcConfigRequirer(
+            charm=self,
+            relation_name=OIDC_PROVIDER_INFO_RELATION,
+        )
```

Comment:
Can't we instantiate this in the `init()` as we do with all library objects? We do it in[ dex as well](https://github.com/canonical/dex-auth-operator/blob/main/src/charm.py#L43-L44)

Reply:
This object is not "passive" as many other objects (e.g. the DexOidcConfigProvider), in fact most "requirers" are handled outside of init.

In this case, we do not want the object to be at init because there we will not handle any exception the library may raise, which can leave the charm in an undesired state. For example, if the relation is not established and we instantiate this object in init, then the charm will go to error right away.

Reply:
Though it is just creating the object, so it should not raise anything at init. I will change it.

Reply:
Everything you wrote makes sense, resolving